### PR TITLE
[Gnosis]: Add order cancellation

### PR DESCRIPTION
**Pull request by evgenyboxer**
# Description

Adds order cancellation.

This PR builds on https://github.com/balancer-labs/frontend-v2/pull/720 so it needs to be merged first.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Make an ERC20 <> ERC20 trade and cancel the order via the transactions activity menu.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] My changes generate no new console warnings
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/components/notifications/Notification.vue [hunk 1]
```diff
@@ -97,10 +97,11 @@ export default defineComponent({
...
       let bgClasses = 'bg-yellow-600 dark:bg-yellow-500';
-      if (props.notification.transactionMetadata?.status === 'confirmed') {
-        bgClasses = props.notification.transactionMetadata.isSuccess
-          ? 'bg-green-500 dark:bg-green-500'
-          : 'bg-red-500 dark:bg-red-500';
+
+      if (props.notification.type === 'success') {
+        bgClasses = 'bg-green-500 dark:bg-green-500';
+      } else if (props.notification.type === 'error') {
+        bgClasses = 'bg-red-500 dark:bg-red-500';
```
Comment (evgenyboxer) [inline src/components/notifications/Notification.vue:104]: using transaction type to get color

File: src/components/navs/AppNav/AppNavActivityBtn/AppNavActivityBtn.vue [hunk 2]
```diff
@@ -70,31 +79,65 @@ export default defineComponent({
...
-    );
+        updateTransaction(orderId, 'order', {
+          status: 'cancelling'
+        });
+      } catch (e) {
+        console.log(e);
+        addNotification({
+          type: 'error',
+          title: t('errorCancellingOrder'),
+          message: e.message
```
Comment (evgenyboxer) [inline src/components/navs/AppNav/AppNavActivityBtn/AppNavActivityBtn.vue:129]: pretty handy to throw this kind of errors when things go wrong... in this instance an attempt to close an order was not successful (perhaps it was already filled)

File: src/composables/useTransactions.ts [hunk 3]
```diff
@@ -22,14 +22,15 @@ import { GnosisTransactionDetails } from './trade/useGnosis';
 
 const WEEK_MS = 86_400_000 * 7;
 // Please update the schema version when making changes to the transaction structure.
-const TRANSACTIONS_SCHEMA_VERSION = '1.1.2';
+const TRANSACTIONS_SCHEMA_VERSION = '1.1.3';
```
Comment (evgenyboxer) [inline src/composables/useTransactions.ts:25]: schema changed

File: src/components/_global/BalLoadingIcon/BalLoadingIcon.vue [hunk 4]
```diff
@@ -31,6 +31,8 @@ export default defineComponent({
           return 'lg';
         case 'sm':
           return 'sm';
+        case 'xs':
+          return 'xs';
```
Comment (evgenyboxer) [inline src/components/_global/BalLoadingIcon/BalLoadingIcon.vue:35]: this works well when the button size is "xs" as well

**Review by garethfuller — APPROVED** [review]

**Pull request merged**